lib/elixir/lib/list.ex

```diff
@@ -872,19 +874,49 @@ defmodule List do
   @doc since: "1.4.0"
   @spec myers_difference(list, list) :: [{:eq | :ins | :del, list}]
   def myers_difference(list1, list2) when is_list(list1) and is_list(list2) do
+    myers_difference_with_diff_script(list1, list2, nil)
+  end
+
+  @doc """
+  Returns a keyword list that represents an *edit script* with nested diffs.
+
+  This is an extension of `myers_difference/2` where a `diff_script` function
+  can be given in case it is desired to compute nested differences. The function
+  may return a list with the inner edit script or `nil` in case there is no
+  such script. The returned inner edit will be under the `:diff` key.
+
+  ## Examples
+
+      iex> List.myers_difference(["a", "db", "c"], ["a", "bc"], &String.myers_difference/2)
+      [eq: ["a"], diff: [del: "d", eq: "b", ins: "c"], del: ["c"]]
+
+  """
+  @doc since: "1.8.0"
+  @spec myers_difference(list, list, (term, term -> script)) :: script
```

Comment:
You removed the `nil` from the `diff_script` with the last commit. `(term, term -> script | nil)`